index.bs

```diff
@@ -3767,46 +3788,57 @@ CountQueuingStrategy({ <var>highWaterMark</var> })</h4>
 <h3 id="queue-with-sizes">Queue-with-Sizes Operations</h3>
 
 The streams in this specification use a "queue-with-sizes" data structure to store queued up values, along with their
-determined sizes. A queue-with-sizes is a List of Records with \[[value]] and \[[size]] fields (although in
-implementations it would of course be backed by a more efficient data structure).
+determined sizes. Various specification objects contain a queue-with-sizes, represented by the object having two paired
+internal slots, always named \[[queue]] and \[[queueTotalSize]]. \[[queue]] is a List of Records with \[[value]] and
+\[[size]] fields, and \[[queueTotalSize]] is a JavaScript {{Number}}, i.e. a double-precision floating point number.
+
+The following abstract operations are used when operating on objects that contain queues-with-sizes, in order to ensure
+that the two internal slots stay synchronized.
 
-A number of abstract operations are specified here to make working with queues-with-sizes more pleasant, and used
-throughout the rest of this standard.
+<p class="warning">Due to the vagaries of floating point arithmetic, the framework specified here, of keeping a running
```

Comment:
Optional: maybe we should say "Due to the limited precision of floating point arithmetic", just to be slightly more specific.